soc/arm/nxp_imx/rt/power_rt11xx.c

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright (c) 2021, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <device.h>
+#include <pm/pm.h>
+#include <fsl_dcdc.h>
+#include <fsl_gpc.h>
+#include <dt-bindings/pm/imx_spc.h>
+#include "power_rt11xx.h"
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+
+/*
+ * NOTE: When multicore support in RT1170/1160 is properly implemented,
+ * power saving will improve when both cores request a transition to a low
+ * power mode
+ */
+
+#ifdef CONFIG_CPU_CORTEX_M7
+#define GPC_CPU_MODE_CTRL GPC_CPU_MODE_CTRL_0
+#elif CONFIG_CPU_CORTEX_M4
+#define GPC_CPU_MODE_CTRL GPC_CPU_MODE_CTRL_1
+#else
+#error "RT11xx power code supports M4 and M7 cores only"
+#endif
+
+/* Configure the set point mappings for Cortex M4 and M7 cores */
+static void gpc_set_core_mappings(void)
+{
+	uint8_t i, j;
+	uint32_t tmp;
+
+#ifdef CONFIG_CPU_CORTEX_M7
+	uint8_t mapping[17][17] = CPU0_COMPATIBLE_SP_TABLE;
```

Comment:
17 is kinda of crypt, may a constant more descriptive for this value ?

Reply:
Fixed this to use a constant